### ISSUE TITLE
how assembly file helloworld2.s is converted in .hex file

### ISSUE THREAD

**Issue by mayank2016** [westerndigitalcorporation/swerv_eh1#23]
Hello SwerV team
I did not get which gcc instruction is used for conversion of assembly file to hex file.

Please help regarding this.

thanks & regards
Mayank

**Comment by aprnath** [comment]
Mayank, please check your source hello.s file. It should not have any 64 bit mnemonics (ld, sd are 64 bit load/stores, and not supported by **-march=rv32imc**).

The Makefile (in $RV_ROOT/tools/Makefile) shows the steps to create the required hex files.

```
program.hex: $(ASM_TEST_DIR)/$(ASM_TEST).s ${RV_ROOT}/configs/snapshots/$(snapshot)/common_defines.vh
    @echo Building $(ASM_TEST)
ifeq ($(shell which $(GCC_PREFIX)-as),)
    @echo " !!! No $(GCC_PREFIX)-as in path, using canned hex files !!"
    cp ${RV_ROOT}/testbench/hex/*.hex .
else
    cp $(ASM_TEST_DIR)/$(ASM_TEST).s .
    $(GCC_PREFIX)-cpp   -I${RV_ROOT}/configs/snapshots/$(snapshot)  $(ASM_TEST).s  > $(ASM_TEST).cpp.s
    $(GCC_PREFIX)-as -march=rv32imc $(ASM_TEST).cpp.s -o $(ASM_TEST).o
    $(GCC_PREFIX)-ld -m elf32lriscv --discard-none -T${RV_ROOT}/testbench/link.ld -o $(ASM_TEST).exe $(ASM_TEST).o
    $(GCC_PREFIX)-objcopy -O verilog --only-section ".data*" --only-section ".rodata*" $(ASM_TEST).exe data.hex
    $(GCC_PREFIX)-objcopy -O verilog --only-section ".text*" --set-start=0x0 $(ASM_TEST).exe program.hex
    $(GCC_PREFIX)-objdump -dS $(ASM_TEST).exe > $(ASM_TEST).dis
    $(GCC_PREFIX)-nm -f posix -C $(ASM_TEST).exe > $(ASM_TEST).tbl
    @echo Completed building $(ASM_TEST)
endif

```

**Comment by mayank2016** [comment]
Thank you sir

Now I have one question
Can I connect any IP with LSU AXI master interface of swerv.
If yes, swerv core can write data in to register of IP like ARM processor
can write data on IP's register through software interface with C API.


Thanks & regards
Mayank

**Comment by aprnath** [comment]
Yes, you should be able to connect any memory addressable IP directly, or via an interconnect matrix to the LSU AXI bus.

There is a [SweRV FPGA build](https://github.com/westerndigitalcorporation/swerv_eh1_fpga) provided by us which has an example that you may be able to reference.

**Comment by mayank2016** [comment]
Sir,
I write C code to read and write in module my_register. C code will execute
inside riscv swerv.
converted C code to assembly from Riscv gcc compiler.
Swerv connected with module my_Register with lsu_AHB interface in a new top
module, add top file in flist.vcs
follow make file and generate program.hex .
Make file vcs-run command in terminal to simulate.
but value is not going to store internal reg q of module with C-api
Reg_write(address,data), i am checking value with $monitor inside
my_register module.

so where can i debug haddrr, hready ,.... signals inside SWERV. actually
where reg_write address value will be store in swerv after assembly
instruction executed? i am stuck.
